framework/classes/attachment.php

```diff
@@ -251,6 +256,13 @@ public function setId($id)
     {
         $path = $this->path();
         $this->attached = $id;
+
+        if(!empty($config['attached_callback']) && is_callable($config['attached_callback'])) {
+            $this->attached = $config['attached_callback']($id);
```

Comment:
$config is undefined here

$this->config['attached_callback']($id, $this->config);